cpp/src/arrow/filesystem/test_util.h

```diff
@@ -168,6 +168,8 @@ class ARROW_TESTING_EXPORT GenericFileSystemTest {
   virtual bool allow_write_file_over_dir() const { return false; }
   // - Whether the filesystem allows reading a directory
   virtual bool allow_read_dir_as_file() const { return false; }
+  // - Whether the filesystem allows moving a file
+  virtual bool allow_move_file() const { return true; }
```

Comment:
We need a way to run these generic tests against the real Azure Data Lake environment.

Reply:
I've added support for running with `AzureFlatNSEnv` and `AzureHierarchicalNSEnv`.